modules/renter/download.go

```diff
@@ -519,3 +521,57 @@ func (r *Renter) DownloadHistory() []modules.DownloadInfo {
 	}
 	return downloads
 }
+
+// ClearDownloadHistory clears the renter's download history based on
+// provided newest and oldest timestamps
+func (r *Renter) ClearDownloadHistory(newest, oldest time.Time) error {
+	if err := r.tg.Add(); err != nil {
+		return err
+	}
+	defer r.tg.Done()
+	r.downloadHistoryMu.Lock()
+	defer r.downloadHistoryMu.Unlock()
+
+	if newest.Before(oldest) && !newest.Equal(time.Unix(0, 0)) {
```

Comment:
I think there might be a bug in this code when searching for a timestamp between existing timestamps.
In my opinion this code is also a bit more complicated than it has to be. I think it would be easier to do it like [this](https://play.golang.org/p/r_6bVd_MU81). I used an `[]int` in the example but apart from that it should be pretty much the same. 
The `main` method tests a bunch of edge cases. If any of those are missing in this PR we should make sure that we add them.